### ISSUE TITLE
[Feature] support creating/dropping extensions

### ISSUE THREAD

**Issue by tdamsma** [olirice/alembic_utils#52]
Would be nice to have a PGExtension class, so alembic can pick up `CREATE EXTENSION` calls. Think it would be a pretty simple addition, or am I missing something?

**Comment by olirice** [comment]
Its a little trickier than you might think depending on how much functionality that class would have

Support for creating a dropping an extension would be easy
```sql
CREATE EXTENSION [ IF NOT EXISTS ] extension_name
    [ WITH ] [ SCHEMA schema_name ]
             [ VERSION version ]
             [ FROM old_version ]
```
and
```sql
DROP EXTENSION [ IF EXISTS ] extension_name [, ...] [ CASCADE | RESTRICT ]
```
with comparisons made through `pg_available_extensions`


But there are several [alter commands](https://www.postgresql.org/docs/current/sql-alterextension.html) and not all of them are reversible (to produce the down migration).

For example
```sql
ALTER EXTENSION name UPDATE [ TO new_version ]
```

there is no equivalent downgrade statement because postgres extensions aren't expected to implement down migrations for the sql bindings. See the migration upgrade files in the [pgsql-http](https://github.com/pramsey/pgsql-http) extension in the form `http--<old_version>--<new_version>.sql`.

Most of those alter statements could likely be ignored/unsupported but `ALTER EXTENSION name SET SCHEMA new_schema` would need to work vs dropping and recreating the extension if the schema associated with `PGExtension` changed.

---

The `ReplaceableEntity` structure but that class is best suited for entities that implement `CREATE OR REPLACE`. I have broken that rule before with some success but its never fun.

With that context, if you'd like to open a PR that implements the minimal `CREATE/DROP/SET SCHEMA` operations with test coverage I'm definitely interested in the feature.

I'll leave this open to track future discussion wrt extensions

**Comment by olirice** [comment]
Thought about this some more

Support for `CREATE EXTENSION` and `DROP EXTENSION` is probably the 99% use case. Additional support for `ALTER` statements could be added in the future in a backwards compatible way so there's no need to get hung up on that now

`alembic_utils.pg_extension.PGExtension` has been added and is available on pypi in `alembic_utils>=0.7.0`

**Comment by tdamsma** [comment]
That was quick, many thanks!
